File: scraper/frontend/src/schema/currency.schema.ts
import z from "zod";

export const currencySchema = z
  .object({
    _id: z.string().optional(),
    id: z.string().optional(),
    name: z.string(),
    code: z.string(),
    symbol: z.string(),
    active: z.boolean(),
    created_at: z.string(),
    updated_at: z.string(),
  })
  .transform((obj) => ({
    ...obj,
    _id: obj._id || obj.id,
    id: obj.id || obj._id,
  }));

export const currenciesSchema = z.object({
  currencies: z.array(currencySchema),
});

export type CurrencyEntity = z.TypeOf<typeof currencySchema>;
export type CurrenciesEntity = z.TypeOf<typeof currenciesSchema>;

File: scraper/backend/app/schemas/currency_schema.py
from pydantic import BaseModel, Field
from typing import Optional


class CurrencyCreate(BaseModel):
    name: str = Field(..., title='Name')
    code: str = Field(..., title='Code')
    symbol: str = Field(..., title='Symbol')
    active: Optional[bool] = True 
    default: Optional[bool] = False
    value: Optional[float] = 0


class CurrencyUpdate(BaseModel):
    name: Optional[str] 
    code: Optional[str] 
    symbol: Optional[str] 
    active: Optional[bool]
    default: Optional[bool]
    value: Optional[float]